File: app/src/main/java/ua/dp/rename/dniprostreets/repo/SnappyMigrator.java
package ua.dp.rename.dniprostreets.repo;

public class SnappyMigrator {

    private SnappyRepository db;

    public SnappyMigrator(SnappyRepository db) {
        this.db = db;
    }

    public void tryMigration() {
        int currentVersion = db.getCacheVersion();
        if (currentVersion < SnappyRepository.CACHE_VERSION)
            migrate(SnappyRepository.CACHE_VERSION);
    }

    private void migrate(int newVersion) {
        if (newVersion == 1) {
            db.clearAll();
            db.setCacheVersion(1);
        }
        //
        if (newVersion == 2) {
            // etc
        }
    }
}

File: app/src/main/java/ua/dp/rename/dniprostreets/repo/SnappyMigrator.kt
package ua.dp.rename.dniprostreets.repo

class SnappyMigrator(private val db: SnappyRepository) {

   fun checkAndTryMigrate() {
      if (db.cacheVersion == SnappyRepository.CACHE_VERSION) return

      db.clearAll()
      db.cacheVersion = SnappyRepository.CACHE_VERSION
   }
}
